lib/std/crypto/rsa.zig

```diff
@@ -0,0 +1,731 @@
+//! RFC8017: Public Key Cryptography Standards #1 v2.2 (PKCS1)
+const std = @import("../std.zig");
+const der = @import("der.zig");
+
+/// Recommend pairing with Sha256.
+pub const Rsa2048 = Rsa(2048);
+/// Recommend pairing with Sha384.
+pub const Rsa3072 = Rsa(3072);
+/// Recommend pairing with Sha512.
+pub const Rsa4096 = Rsa(4096);
+
+pub fn Rsa(comptime modulus_bits: usize) type {
+    // To keep implementation simpler. May be lifted in the future.
+    if (modulus_bits & 8 != 0) @compileError("modulus_bits must be divisible by 8");
+    return struct {
+        const Modulus = std.crypto.ff.Modulus(modulus_bits);
+        const Fe = Modulus.Fe;
+        pub const modulus_len = std.math.divCeil(usize, modulus_bits, 8) catch unreachable;
+
+        pub const PublicKey = struct {
+            /// `n`
+            modulus: Modulus,
+            /// `e`
+            public_exponent: Fe,
+
+            pub fn fromBytes(mod: []const u8, exp: []const u8) !PublicKey {
+                if (mod.len != modulus_len) return error.InvalidModulus;
+                const modulus = try Modulus.fromBytes(mod, .big);
+                const public_exponent = try Fe.fromBytes(modulus, exp, .big);
```

Comment:
The current code makes sure that the modulus is at least 512 bits, and doesn't even try to parse exponents that would be too large. Maybe it would be good to keep that.

Reply:
The new code allows a user to pick `modulus_len` at compile time. In the spirit of not allowing users to do insecure things I'll add a compile time check to enforce >= 512 bits.